lsat/caveat.go

```diff
@@ -0,0 +1,145 @@
+package lsat
+
+// this whole file is just copied from lightning labs' aperture
```

Comment:
aperture's code isn't exposed as a module so I just copied most of it. It should be fine and relatively static since the caveats stuff is pretty simple